### ISSUE TITLE
Syncthing (other things?) does not notice coderoot changes

### ISSUE THREAD

**Issue by labboy0276** [kalabox/kalabox#377]
OK

So I tried to do a UCSF setup with the new hotness of changing the coderoot, etc.

However, when I change me coderoot and / or my codedir via my kalabox.json:

```
  "codeRoot": "/home/john/kalabox/apps/ucsf/code/docroot",
  "codeDir": "docroot",
```

I get

```
nfo: DOCKER =>  Ad hoc container started. ac8c6b7231812e9a2142f59db9cb33cd6abef68950190bd1ddb2ec6b37178af3
info: DOCKER =>  Removing ad hoc container. ac8c6b7231812e9a2142f59db9cb33cd6abef68950190bd1ddb2ec6b37178af3
EVENTS =>  Error dispatching event listeners [pre-start].
Error: cp: cannot stat `/src/code/.stignore': No such file or directory

    at MemoryDuplexStream.<anonymous> (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:544:24)
    at MemoryDuplexStream.emit (events.js:107:17)
    at readableAddChunk (_stream_readable.js:163:16)
    at MemoryDuplexStream.Readable.push (_stream_readable.js:126:10)
    at MemoryDuplexStream._write (/home/john/kb2/node_modules/kalabox-engine-docker/node_modules/memorystream/index.js:150:14)
    at doWrite (_stream_writable.js:301:12)
    at writeOrBuffer (_stream_writable.js:288:5)
    at MemoryDuplexStream.Writable.write (_stream_writable.js:217:11)
    at IncomingMessage.<anonymous> (/home/john/kb2/node_modules/kalabox-engine-docker/node_modules/dockerode/node_modules/docker-modem/lib/modem.js:247:16)
    at IncomingMessage.emit (events.js:104:17)
    at emitReadable_ (_stream_readable.js:424:10)
    at emitReadable (_stream_readable.js:418:7)
    at readableAddChunk (_stream_readable.js:174:11)
    at IncomingMessage.Readable.push (_stream_readable.js:126:10)
    at HTTPParser.parserOnBody (_http_common.js:132:22)
    at Socket.socketOnData (_http_client.js:317:20)
From previous event:
    at Object.result.wait (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:539:20)
    at /home/john/kb2/node_modules/kalabox-engine-docker/docker.js:612:18
From previous event:
    at Object.queryData (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:610:6)
From previous event:
    at Object.exports.queryData (/home/john/kb2/lib/engine.js:628:27)
    at /home/john/kb2/plugins/kalabox-syncthing/events.js:121:27
From previous event:
    at Object.<anonymous> (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:692:25)
From previous event:
    at Object.use (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:691:6)
From previous event:
    at /home/john/kb2/lib/engine.js:566:29
From previous event:
    at Object.exports.use (/home/john/kb2/lib/engine.js:565:4)
    at /home/john/kb2/plugins/kalabox-syncthing/events.js:120:25
From previous event:
    at /home/john/kb2/plugins/kalabox-syncthing/events.js:96:10
    at /home/john/kb2/lib/core/events.js:131:9
    at /home/john/kb2/lib/core/events.js:128:20
From previous event:
    at EventEngine._dispatch (/home/john/kb2/lib/core/events.js:126:18)
    at /home/john/kb2/lib/core/events.js:106:19
From previous event:
    at EventEngine.emit (/home/john/kb2/lib/core/events.js:98:21)
From previous event:
    at Object.exports.emit (/home/john/kb2/lib/core/events.js:196:4)
    at /home/john/kb2/lib/app.js:500:24
From previous event:
    at Object.exports.start (/home/john/kb2/lib/app.js:499:4)
    at Object.task.func (/home/john/kb2/plugins/kalabox-core/tasks/start.js:14:18)
    at Object.exports.run (/home/john/kb2/lib/core/tasks.js:773:15)
    at processTask (/home/john/kb2/bin/kbox.js:268:16)
    at /home/john/kb2/bin/kbox.js:330:19
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

For the hell of it, the file was in /code/docroot so I did move it to /code and it did start.

However, it should recognize that the codedir / coderoot is different.

**Comment by bcauldwell** [comment]
Don't change the codeDir, only change the codeRoot.

On Thu, Aug 13, 2015 at 9:27 AM, John Ouellet <notifications@github.com>
wrote:

> OK
>
> So I tried to do a UCSF setup with the new hotness of changing the
> coderoot, etc.
>
> However, when I change me coderoot and / or my codedir via my kalabox.json:
>
>   "codeRoot": "/home/john/kalabox/apps/ucsf/code/docroot",
>   "codeDir": "docroot",
>
> I get
>
> nfo: DOCKER =>  Ad hoc container started. ac8c6b7231812e9a2142f59db9cb33cd6abef68950190bd1ddb2ec6b37178af3
> info: DOCKER =>  Removing ad hoc container. ac8c6b7231812e9a2142f59db9cb33cd6abef68950190bd1ddb2ec6b37178af3
> EVENTS =>  Error dispatching event listeners [pre-start].
> Error: cp: cannot stat `/src/code/.stignore': No such file or directory
>
>     at MemoryDuplexStream.<anonymous> (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:544:24)
>     at MemoryDuplexStream.emit (events.js:107:17)
>     at readableAddChunk (_stream_readable.js:163:16)
>     at MemoryDuplexStream.Readable.push (_stream_readable.js:126:10)
>     at MemoryDuplexStream._write (/home/john/kb2/node_modules/kalabox-engine-docker/node_modules/memorystream/index.js:150:14)
>     at doWrite (_stream_writable.js:301:12)
>     at writeOrBuffer (_stream_writable.js:288:5)
>     at MemoryDuplexStream.Writable.write (_stream_writable.js:217:11)
>     at IncomingMessage.<anonymous> (/home/john/kb2/node_modules/kalabox-engine-docker/node_modules/dockerode/node_modules/docker-modem/lib/modem.js:247:16)
>     at IncomingMessage.emit (events.js:104:17)
>     at emitReadable_ (_stream_readable.js:424:10)
>     at emitReadable (_stream_readable.js:418:7)
>     at readableAddChunk (_stream_readable.js:174:11)
>     at IncomingMessage.Readable.push (_stream_readable.js:126:10)
>     at HTTPParser.parserOnBody (_http_common.js:132:22)
>     at Socket.socketOnData (_http_client.js:317:20)
> From previous event:
>     at Object.result.wait (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:539:20)
>     at /home/john/kb2/node_modules/kalabox-engine-docker/docker.js:612:18
> From previous event:
>     at Object.queryData (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:610:6)
> From previous event:
>     at Object.exports.queryData (/home/john/kb2/lib/engine.js:628:27)
>     at /home/john/kb2/plugins/kalabox-syncthing/events.js:121:27
> From previous event:
>     at Object.<anonymous> (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:692:25)
> From previous event:
>     at Object.use (/home/john/kb2/node_modules/kalabox-engine-docker/docker.js:691:6)
> From previous event:
>     at /home/john/kb2/lib/engine.js:566:29
> From previous event:
>     at Object.exports.use (/home/john/kb2/lib/engine.js:565:4)
>     at /home/john/kb2/plugins/kalabox-syncthing/events.js:120:25
> From previous event:
>     at /home/john/kb2/plugins/kalabox-syncthing/events.js:96:10
>     at /home/john/kb2/lib/core/events.js:131:9
>     at /home/john/kb2/lib/core/events.js:128:20
> From previous event:
>     at EventEngine._dispatch (/home/john/kb2/lib/core/events.js:126:18)
>     at /home/john/kb2/lib/core/events.js:106:19
> From previous event:
>     at EventEngine.emit (/home/john/kb2/lib/core/events.js:98:21)
> From previous event:
>     at Object.exports.emit (/home/john/kb2/lib/core/events.js:196:4)
>     at /home/john/kb2/lib/app.js:500:24
> From previous event:
>     at Object.exports.start (/home/john/kb2/lib/app.js:499:4)
>     at Object.task.func (/home/john/kb2/plugins/kalabox-core/tasks/start.js:14:18)
>     at Object.exports.run (/home/john/kb2/lib/core/tasks.js:773:15)
>     at processTask (/home/john/kb2/bin/kbox.js:268:16)
>     at /home/john/kb2/bin/kbox.js:330:19
>     at processImmediate [as _immediateCallback] (timers.js:367:17)
>
> For the hell of it, the file was in /code/docroot so I did move it to
> /code and it did start.
>
> However, it should recognize that the codedir / coderoot is different.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/377>.
>



-- 
- Benjamin Cauldwell

**Comment by labboy0276** [comment]
Even if I dont change codeDir it still does that.

**Comment by pirog** [comment]
This should be resolved in the 0.11.0 release on Friday
